skxray/core/fitting/xrf_model.py

```diff
@@ -1233,3 +1234,164 @@ def _get_activated_line(incident_energy, elemental_line):
                 continue
             line_list.append(str(element)+'_'+str(line_name))
         return line_list
+
+
+def fit_per_line_nnls(row_num, data,
+                      matv, param, use_snip):
+    """
+    Fit experiment data for a given row using nnls algorithm.
+    Parameters
+    ----------
+    row_num : int
+        which row to fit
+    data : array
+        selected one row of experiment spectrum
+    matv : array
+        matrix for regression analysis
+    param : dict
+        fitting parameters
+    use_snip : bool
+        use snip algorithm to remove background or not
+
+    Returns
+    -------
+    array :
+        fitting values for all the elements at a given row. Background is
+        calculated as a summed value. Also residual is included.
+    """
+    logger.info('Row number at {}'.format(row_num))
```

Comment:
it is used to show which row is processed during a multiple processing. I actually didn't see other way how to output row number.